MantidQt/CustomInterfaces/src/Tomography/TomoToolConfigDialogSavu.cpp

```diff
@@ -1,23 +1,90 @@
-#include "MantidAPI/AlgorithmManager.h"
-#include "MantidAPI/ITableWorkspace.h"
-#include "MantidQtCustomInterfaces/Tomography/TomographyIfaceViewQtGUI.h"
-
-#include <boost/lexical_cast.hpp>
+#include "MantidQtCustomInterfaces/Tomography/TomoToolConfigDialogSavu.h"
+#include "MantidQtCustomInterfaces/Tomography/TomoReconToolsUserSettings.h"
 
+#include <MantidAPI/TableRow.h>
+#include <MantidAPI/WorkspaceFactory.h>
 #include <Poco/String.h>
-
+#include <json/reader.h>
+#include <MantidAPI/NotebookWriter.h>
 #include <QFileDialog>
 #include <QMessageBox>
+#include <MantidAPI/AlgorithmManager.h>
 
 using namespace Mantid::API;
 
 namespace MantidQt {
 namespace CustomInterfaces {
 
+TomoToolConfigDialogSavu::TomoToolConfigDialogSavu(QWidget *parent)
+    : QMainWindow(parent) {
+  // TODO init the 545231453t54 variables
+  m_availPlugins = Mantid::API::WorkspaceFactory::Instance().createTable();
+  m_availPlugins->addColumns("str", "name", 4);
+  m_currPlugins = Mantid::API::WorkspaceFactory::Instance().createTable();
+  m_currPlugins->addColumns("str", "name", 4);
+}
+
+int TomoToolConfigDialogSavu::executeQt() {
+  this->show();
+  QEventLoop el;
+  connect(this, SIGNAL(destroyed()), &el, SLOT(quit()));
+  return el.exec();
+}
+
+void TomoToolConfigDialogSavu::initialiseDialog() {
+  throw Mantid::Kernel::Exception::NotImplementedError(
+      "SAVU interface not implemented");
+}
+void TomoToolConfigDialogSavu::setupMethodSelected() {
+  throw Mantid::Kernel::Exception::NotImplementedError(
+      "SAVU interface not implemented");
+}
+void TomoToolConfigDialogSavu::setupToolSettingsFromPaths() {
+  throw Mantid::Kernel::Exception::NotImplementedError(
+      "SAVU interface not implemented");
+}
+
+void TomoToolConfigDialogSavu::setupDialogUi() {
+  m_savuUi.setupUi(this);
+  initSavuWindow();
+  this->setWindowModality(Qt::ApplicationModal);
+}
+
+void TomoToolConfigDialogSavu::initSavuWindow() {
+  // geometry, etc. niceties
+  // on the left (just plugin names) 1/2, right: 2/3
+  QList<int> sizes;
```

Comment:
Does a `QList` have a reserve so we can reserve both spaces before pushing back into them